internal/db/db_blueprints.go

```diff
@@ -27,6 +45,46 @@ const (
 	sqlDeleteBlueprint = `DELETE FROM blueprints WHERE id = $1 AND org_id = $2 AND account_number = $3`
 )
 
+func (db *dB) GetBlueprintComposes(orgId string, blueprintId uuid.UUID, since time.Duration, limit, offset int, ignoreImageTypes []string) ([]ComposeEntry, int, error) {
+	ctx := context.Background()
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer conn.Release()
+	result, err := conn.Query(ctx, sqlGetBlueprintComposes, orgId, blueprintId, since, ignoreImageTypes, limit, offset)
+
+	if err != nil {
+		return nil, 0, err
+	}
+
+	var composes []ComposeEntry
+	for result.Next() {
+		var jobId uuid.UUID
+		var version int
+		var request json.RawMessage
+		var createdAt time.Time
+		var imageName *string
+		var clientId *string
+		err = result.Scan(&jobId, &version, &request, &createdAt, &imageName, &clientId)
+		if err != nil {
+			return nil, 0, err
+		}
+		composes = append(composes, newComposeEntry(jobId, blueprintId, version, request, createdAt, imageName, clientId))
+	}
```

Comment:
Please scan into the result value directly, not via a temporary variables and the unnecessary function `newComposeEntry`.

```
var ce []ComposeEntry
...
err = result.Scan(
  &ce.Id,
  &ce.BlueprintId,
)
```

I intentionally created a multi-line form because I believe it is ultimately better.

Reply:
It's an array, so it might not be easey, but we could append first and then Scan into that appended var :shrug:

Reply:
Yeah that slice was a typo in my example, what I meant was:

```
var ce ComposeEntry
...
err = result.Scan(
  &ce.Id,
  &ce.BlueprintId,
)
...
result = append(result, ce)
```